src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -171,28 +177,28 @@ private void BuildSwitchLabels(SyntaxList<SwitchLabelSyntax> labelsSyntax, Binde
                 switch (labelSyntax.Kind())
                 {
                     case SyntaxKind.CaseSwitchLabel:
-                    // Bind the switch expression and the switch case label expression, but do not report any diagnostics here.
-                    // Diagnostics will be reported during binding.
-                    var caseLabel = (CaseSwitchLabelSyntax)labelSyntax;
-                    Debug.Assert(caseLabel.Value != null);
-                    DiagnosticBag tempDiagnosticBag = DiagnosticBag.GetInstance();
+                        // Bind the switch expression and the switch case label expression, but do not report any diagnostics here.
+                        // Diagnostics will be reported during binding.
+                        var caseLabel = (CaseSwitchLabelSyntax)labelSyntax;
+                        Debug.Assert(caseLabel.Value != null);
+                        DiagnosticBag tempDiagnosticBag = DiagnosticBag.GetInstance();
 
-                    var boundLabelExpression = sectionBinder.BindValue(caseLabel.Value, tempDiagnosticBag, BindValueKind.RValue);
+                        var boundLabelExpression = sectionBinder.BindValue(caseLabel.Value, tempDiagnosticBag, BindValueKind.RValue);
 
-                    if ((object)switchGoverningType == null)
-                    {
-                        switchGoverningType = this.BindSwitchExpression(_switchSyntax.Expression, this, tempDiagnosticBag).Type;
-                    }
+                        if ((object)switchGoverningType == null)
+                        {
+                            switchGoverningType = this.BindSwitchExpression(_switchSyntax.Expression, this, tempDiagnosticBag).Type;
+                        }
 
-                    boundLabelExpression = ConvertCaseExpression(switchGoverningType, labelSyntax, boundLabelExpression, sectionBinder, ref boundLabelConstantOpt, tempDiagnosticBag);
+                        boundLabelExpression = ConvertCaseExpression(switchGoverningType, labelSyntax, boundLabelExpression, sectionBinder, ref boundLabelConstantOpt, tempDiagnosticBag);
 
-                    tempDiagnosticBag.Free();
+                        tempDiagnosticBag.Free();
                         break;
 
                     case SyntaxKind.DefaultSwitchLabel:
                         break;
                     case SyntaxKind.CasePatternSwitchLabel:
-                        throw new NotImplementedException();
+                        continue;
```

Comment:
This occurs because we parse new trees (with a syntax error) in old language levels. This is exercised by `PatternParsingTexts.CasePatternVersusFeatureFlag`


---
In reply to: [64947533](https://github.com/dotnet/roslyn/pull/11485#discussion_r64947533) [](ancestors = 64947533)